dali/operators/random/noise/shot_noise.h

```diff
@@ -21,27 +21,31 @@
 #include "dali/operators/random/rng_base_cpu.h"
 
 #define DALI_SHOT_NOISE_TYPES \
-  (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float)
+  uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float
 
 namespace dali {
 
 template <typename Backend, typename T>
-class shot_noise_impl {
+class ShotNoiseImpl {
  public:
-  using Dist =
+  using DistType =
       typename std::conditional_t<std::is_same<Backend, GPUBackend>::value,
                                   curand_poisson_dist,
                                   std::poisson_distribution<uint32_t>>;
 
-  DALI_HOST_DEV explicit shot_noise_impl(float factor = 12)
+  DALI_HOST_DEV explicit ShotNoiseImpl(float factor = 12)
       : factor_{factor}, inv_factor_{1.0f / factor_} {}
 
   template <typename Generator>
-  DALI_HOST_DEV inline T operator()(T in_val, Generator& g) {
+  DALI_HOST_DEV uint32_t Generate(T in_val, Generator& g) {
     if (factor_ == 0.0f)
       return in_val;
-    Dist dist(cuda_max<float>(0.0f, in_val * inv_factor_));
-    return ConvertSat<T>(dist(g) * factor_);
+    DistType dist(cuda_max<float>(0.0f, in_val * inv_factor_));
+    return dist(g);
+  }
+
+  DALI_HOST_DEV void Apply(T &output, T input, uint32_t n) {
+    output = ConvertSat<T>(n * factor_);
   }
```

Comment:
How you envision applying the same noise to different channels as the generation also depends on the pixel value?